packages/fxa-graphql-api/src/gql/dto/input/sign-in.ts

```diff
@@ -41,7 +41,7 @@ export class SignInInput {
   public clientMutationId?: string;
 
   @Field()
-  public authPW!: hexstring;
```

Comment:
Curious why this had to change?

Reply:
@vbudhram Alright, circling back on this. I rolled back the changes surrounding changing `hexstring` to `string`, and I was able to trigger the error again. You can see the resulting error here: https://app.circleci.com/pipelines/github/mozilla/fxa/37116/workflows/45494a65-7b3e-4ff8-96c4-ad2cf40732c7/jobs/397513.

I did some digging and it appears that when the `isolatedModules` option was added to the jest-config to `ts-jest`, the error started popping up. It seems that any type defs in other workspace result in an error for the e2e tests.

I should note that adding the `isolateModules:true` to the `ts-jest` config made a big difference when it comes to the memory footprint needed to run tests. We can see this now that the `--logHeap` option is being passed to jest.

Before Optimization:
```
➜  fxa-graphql-api git:(FXA-6266) ✗ yarn workspace fxa-graphql-api test
ts-jest[versions] (WARN) Version 27.5.1 of jest installed has not been tested with ts-jest. If you're experiencing issues, consider using a supported version (>=29.0.0 <30.0.0-0). Please do not report issues in ts-jest if you are using unsupported versions.
 PASS  src/gql/account.resolver.spec.ts (876 MB heap size)
 PASS  src/gql/session.resolver.spec.ts (927 MB heap size)
 PASS  src/database/database.service.spec.ts (877 MB heap size)
 PASS  src/backend/profile-client.service.spec.ts (897 MB heap size)
 PASS  src/auth/session-token.strategy.spec.ts (912 MB heap size)
```

After Optimization:
```
➜  fxa-graphql-api git:(FXA-6266) ✗ yarn workspace fxa-graphql-api test
ts-jest[versions] (WARN) Version 27.5.1 of jest installed has not been tested with ts-jest. If you're experiencing issues, consider using a supported version (>=29.0.0 <30.0.0-0). Please do not report issues in ts-jest if you are using unsupported versions.
 PASS  src/gql/account.resolver.spec.ts (151 MB heap size)
 PASS  src/gql/session.resolver.spec.ts (210 MB heap size)
 PASS  src/database/database.service.spec.ts (162 MB heap size)
 PASS  src/backend/profile-client.service.spec.ts (186 MB heap size)
 PASS  src/auth/session-token.strategy.spec.ts (194 MB heap size)
 ```

After several passes, I finally found a way to preserve the `hexstring` type but still have the e2e tests pass with `isolatedModules:true` configured. The issue is that we now have to give nestjs a hint as to what the underlying nestjs field type really is. This sort of feels like a bug in nestjs because I don't see why `isolatedModules` should behave differently. The stack trace seems to indicate that the reflection code runs into an issue when executed from the context of a TestingModule, as TS compilation encounters no issue, and unit tests run fine.

Just in case the CI job state disappears here is the exact error and callstack:

```Undefined type error. Make sure you are providing an explicit type for the "uid" of the "SignedInAccountPayload" class.

  	38 |
  	39 | @ObjectType()
	> 40 | export class SessionReauthedAccountPayload extends OmitType(
     	|                                                	^
  	41 |   SignedInAccountPayload,
  	42 |   ['sessionToken']
  	43 | ) {}

  	at Object.reflectTypeFromMetadata (../../../node_modules/@nestjs/graphql/dist/utils/reflection.utilts.js:15:15)
  	at applyMetadataFn (../../../node_modules/@nestjs/graphql/dist/decorators/field.decorator.js:21:79)
  	at ../../../node_modules/@nestjs/graphql/dist/schema-builder/storages/lazy-metadata.storage.js:35:44
      	at Array.forEach (<anonymous>)
  	at LazyMetadataStorageHost.load (../../../node_modules/@nestjs/graphql/dist/schema-builder/storages/lazy-metadata.storage.js:35:26)
  	at Object.getFieldsAndDecoratorForType (../../../node_modules/@nestjs/graphql/dist/schema-builder/utils/get-fields-and-decorator.util.js:16:49)
  	at OmitType (../../../node_modules/@nestjs/graphql/dist/type-helpers/omit-type.helper.js:10:74)
  	at Object.<anonymous> (gql/dto/payload/signed-in-account.ts:40:52)
```

Reply:
Alright, circling back on this. I rolled back the changes surrounding changing `hexstring` to `string`, and I was able to triggered the error again. You can see the resulting error here: https://app.circleci.com/pipelines/github/mozilla/fxa/37116/workflows/45494a65-7b3e-4ff8-96c4-ad2cf40732c7/jobs/397513.

I did some digging and it appears that when the `isolatedModules` option was added to the jest-config to `ts-jest`, the error started popping up. It seems that any type defs in other workspace result in an error for the e2e tests.

I should note that adding the `isolateModules:true` to the `ts-jest` config made a big difference when it comes to the memory footprint needed to run tests. We can see this now that the `--logHeap` option is being passed to jest.

Before Optimization:
```
➜  fxa-graphql-api git:(FXA-6266) ✗ yarn workspace fxa-graphql-api test
ts-jest[versions] (WARN) Version 27.5.1 of jest installed has not been tested with ts-jest. If you're experiencing issues, consider using a supported version (>=29.0.0 <30.0.0-0). Please do not report issues in ts-jest if you are using unsupported versions.
 PASS  src/gql/account.resolver.spec.ts (876 MB heap size)
 PASS  src/gql/session.resolver.spec.ts (927 MB heap size)
 PASS  src/database/database.service.spec.ts (877 MB heap size)
 PASS  src/backend/profile-client.service.spec.ts (897 MB heap size)
 PASS  src/auth/session-token.strategy.spec.ts (912 MB heap size)
```

After Optimization:
```
➜  fxa-graphql-api git:(FXA-6266) ✗ yarn workspace fxa-graphql-api test
ts-jest[versions] (WARN) Version 27.5.1 of jest installed has not been tested with ts-jest. If you're experiencing issues, consider using a supported version (>=29.0.0 <30.0.0-0). Please do not report issues in ts-jest if you are using unsupported versions.
 PASS  src/gql/account.resolver.spec.ts (151 MB heap size)
 PASS  src/gql/session.resolver.spec.ts (210 MB heap size)
 PASS  src/database/database.service.spec.ts (162 MB heap size)
 PASS  src/backend/profile-client.service.spec.ts (186 MB heap size)
 PASS  src/auth/session-token.strategy.spec.ts (194 MB heap size)
 ```

After several passes, I finally found a way to preserve the `hexstring` type but still have the e2e tests pass with `isolatedModules:true` configured. The issue is that we now have to give nestjs a hint as to what the underlying nestjs filed type really is. This sort of feels like a bug in nestjs because I don't see why `isolatedModules` should behave differently. The stack trace seems to indicate that the reflection code runs into an issue when executed from the context of a TestingModule, as ts compilation encounters no issue, and unit tests run fine.

Just in case the ci job state disappears here is the Exact error and callstack:

```Undefined type error. Make sure you are providing an explicit type for the "uid" of the "SignedInAccountPayload" class.

      38 |
      39 | @ObjectType()
    > 40 | export class SessionReauthedAccountPayload extends OmitType(
         |                                                    ^
      41 |   SignedInAccountPayload,
      42 |   ['sessionToken']
      43 | ) {}

      at Object.reflectTypeFromMetadata (../../../node_modules/@nestjs/graphql/dist/utils/reflection.utilts.js:15:15)
      at applyMetadataFn (../../../node_modules/@nestjs/graphql/dist/decorators/field.decorator.js:21:79)
      at ../../../node_modules/@nestjs/graphql/dist/schema-builder/storages/lazy-metadata.storage.js:35:44
          at Array.forEach (<anonymous>)
      at LazyMetadataStorageHost.load (../../../node_modules/@nestjs/graphql/dist/schema-builder/storages/lazy-metadata.storage.js:35:26)
      at Object.getFieldsAndDecoratorForType (../../../node_modules/@nestjs/graphql/dist/schema-builder/utils/get-fields-and-decorator.util.js:16:49)
      at OmitType (../../../node_modules/@nestjs/graphql/dist/type-helpers/omit-type.helper.js:10:74)
      at Object.<anonymous> (gql/dto/payload/signed-in-account.ts:40:52)
```

Reply:
Wow that is a huge increase in memory improvement. I think we should go with the version that reduces memory. We currently don't have any joi type validation in graphql and that is probably better tackled in another issue.